server/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java

```diff
@@ -190,12 +205,21 @@ public double getPbFrac() {
      *         serialising the value to JSON.
      */
     public String getStringRep() {
-        if (size <= 0) {
-            return String.valueOf(size);
+        if (sizeInBytes <= 0) {
+            return String.valueOf(sizeInBytes);
+        }
+        long numUnits = sizeInBytes / desiredUnit.toBytes(1);
+        long residue = sizeInBytes % desiredUnit.toBytes(1);
+        if (residue == 0) {
+            return numUnits + desiredUnit.getSuffix();
+        } else {
+            return sizeInBytes + BYTES.getSuffix();
```

Comment:
I see, if there is a reminder, we always return a string representation as bytes, which do not have a reminder. Clever, it actually implements what written in the javadoc :)

Reply:
Yeah my goal here is backward compatibility. If designed from scratch, this would not make a lot of sense TBH.